### DISCUSSION TITLE
Should we automatically close old issues?

### DISCUSSION THREAD

**Discussion by AnHeuermann** [OpenModelica/OpenModelica#9605] [Polls]
We have over 1.5k issues. I'm assuming not all issues are relevant any more.

A lot of those issues are migrated from [trac.openmodelica.org](https://trac.openmodelica.org/OpenModelica/) and look like https://github.com/OpenModelica/OpenModelica/issues/1232.
Systematically checking if each and every issue is still reproduceable would be great, but consuming way more resources than we have.
Let's say we check 500 issues times 5 minutes per ticket and we end up with over 40 hours of work. And 5 minutes per ticket is very conservative. So there is no way to do this.
 
We could use some automation to mark all stale issues and close them after some grace period or set some manually to never be closed. Important issues will be opened again and if no one encounters the issue again it probably wasn't so important.
But as a user it could feel punishing to fight the automation to keep your very important issue open, because nobody had time to look at it yet.

-------

PS: I'm just curious what others think, this poll won't mean that something has to change in one way or the other.

**Comment by perost** [comment]
I vote no, at least I don't want the old frontend issues closed since a lot of them are still valid and something I look at from time to time. Most of those issues only take a few seconds to check, just paste the example model into a file and see if it still gives the same error.

But of course there are probably tickets that are more complicated and/or not explained well enough to easily reproduce, so some of them should probably be closed. But I don't think it's really that much work to go through them and check which ones are still relevant, I usually do it about once a year for the frontend tickets. I don't think anyone checks the backend ones though, so maybe we could close a lot of the tickets if someone did that.

**Comment by casella** [comment]
@AnHeuermann I've been handling tickets for a few years. Basically, all of them since 2018 or something. It's a dirty job, but someone has to do it. Can also be fun, in some cases, some people play Sudoku, I play whack-a-ticket with OMC issues 😃 

It is probably true that the majority of the 1.5k old tickets are obsolete, but there are most likely a few hundreds of them which are extremely valuable, because each of them pointed out a problem and involved some non-trivial analysis work, maybe even provided a nice MWE to test it. Very conservative estimate: say we have 200 valuable tickets with a couple hours worth of analysis work behind each of them on average, it gives us 400 hrs of skilled person's time. That's 3 PMs, worth 10-20.000 EUR at current labour cost. Could actually be much more than that. You don't want to throw away this potential value lightly.

Also, the idea of migrating all the information automatically from trac to GIT was eventually abandoned, because it was simply too complicated.

We did comb through old tickets on trac once a few years ago, and managed to close many of them. That's no big deal, if you spread the job over multiple week or months and possibly over multiple people. In this case, I would expect that most of these issues are clearly obsolete and can be dismissed in less than 5 mins, while the valuable ones can be properly migrated from trac to GIT by adding the proper references and info to the GIT ticket. For sure this will take much less time that re-doing the relevant ones from scratch.

I'll try to do that once we manage to get through the 1.20.0 release and the end-of-october DFD deadlines.
